flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironmentImpl.java

```diff
@@ -125,7 +125,10 @@ public void prepare(Config config) throws Exception {
         zookeeper = null;
         brokers.clear();
 
-        zookeeper = new TestingServer(-1, tmpZkDir);
+        synchronized (KafkaTestEnvironmentImpl.class) {
+            zookeeper = new TestingServer(-1, tmpZkDir);
+        }
+
```

Comment:
This will probably not do anything. The issue is that multiple parallel processes are competing for the port on the same machine. Kafka tests are only executed non-parallel afaik, so there will never be 2 threads competing for the same port.

Reply:
Thank you for your advice @AHeise , I implemented a file lock to avoid race condition among multiple unit test processes.